MatrixSDK/Data/MXRoomSummaryUpdater.m

```diff
@@ -756,4 +755,42 @@ - (BOOL)isEventUserProfileChange:(MXEvent*)event
     return event.isUserProfileChange;
 }
 
+- (BOOL)isMembershipEventJoinOrInvite:(MXEvent*)event forUserId:(NSString*)userId
+{
+    if (event.eventType != MXEventTypeRoomMember)
+    {
+        return NO;
+    }
+    
+    NSString *eventUserId = event.stateKey;
+    
+    BOOL isMembershipEventAffectingUser = eventUserId && userId && [userId isEqualToString:eventUserId];
```

Comment:
Does the `eventUserId && userId &&` doing anything here? `[userId isEqualToString:eventUserId]` would only evaluate to true if they're both non-nil anyway?